### ISSUE TITLE
Modify Library

### ISSUE THREAD

**Issue by mateo2181** [jbaysolutions/vue-grid-layout#253]
Hello! I am trying to modify library to create a swap function.
I have changed vue.config.js to this:
![selection_003](https://user-images.githubusercontent.com/12451246/48855023-06ce6d80-ed92-11e8-90b8-77119899e6ba.png)
I run "npm run build-lib" and works.
But when I use the library I got this error: 

[Vue warn]: Failed to mount component: template or render function not defined.
found in
---> <GridLayout>
       <Grid> at src/components/Grid.vue
         <App> at src/App.vue
           <Root>

Someone know how can I fix this ? 

Thanks!

**Comment by rebz** [comment]
@mateo2181 - I know it's been a couple years, did you manage to get swapping to work or did you use a different library?

**Comment by Jacks349** [comment]
I'm looking for a way to swap items too, did anyone manage to do that? Basically i would like to be able to swap the items horizontally too instead of vertically only, don't know if and how it's doable

**Comment by rebz** [comment]
> I'm looking for a way to swap items too, did anyone manage to do that? Basically i would like to be able to swap the items horizontally too instead of vertically only, don't know if and how it's doable

I just built a custom solution, no packages. Struggled to find a decent drag and drop package with touch support that was actively maintained.

**Comment by Jacks349** [comment]
> > I'm looking for a way to swap items too, did anyone manage to do that? Basically i would like to be able to swap the items horizontally too instead of vertically only, don't know if and how it's doable
> 
> I just built a custom solution, no packages. Struggled to find a decent drag and drop package with touch support that was actively maintained.

Thanks for your answer! So you didn't use this library in the end? It's a shame because except for that, vue-grid-layout has everything i needed

**Comment by mateo2181** [comment]
I created this library to be able to swap elements https://github.com/mateo2181/vue-swappable.

**Comment by rebz** [comment]
> > > I'm looking for a way to swap items too, did anyone manage to do that? Basically i would like to be able to swap the items horizontally too instead of vertically only, don't know if and how it's doable
> > 
> > 
> > I just built a custom solution, no packages. Struggled to find a decent drag and drop package with touch support that was actively maintained.
> 
> Thanks for your answer! So you didn't use this library in the end? It's a shame because except for that, vue-grid-layout has everything i needed

Correct, all custom solution. I had tried a variety of packages out there but I was building a planogram builder for vending machines and there were too many variables required... a few of which are:
- Variable height and resizable rows
- Swappable items
- Swappable rows
- Collision detection

> I created this library to be able to swap elements https://github.com/mateo2181/vue-swappable.

Interesting, I’m working on another drag and drop project with touch support for a custom video surveillance dashboard. I once looked at Draggable.JS but saw it was no longer maintained. I will have to check this out.

**Comment by Jacks349** [comment]
Hello everyone, i gave it a try to implement my own swapping [here](https://github.com/jbaysolutions/vue-grid-layout/issues/604)
